server/src/main/java/org/opensearch/common/metadata/MetadataManager.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.common.metadata;
+
+import java.io.IOException;
+import java.util.Map;
+
+import org.apache.lucene.codecs.CodecUtil;
+import org.apache.lucene.store.BufferedChecksumIndexInput;
+import org.apache.lucene.store.ChecksumIndexInput;
+import org.apache.lucene.store.IndexInput;
+import org.apache.lucene.store.IndexOutput;
+
+/**
+ * Manages Metadata by adding version, codec in header and checksum in footer.
+ * @param <T> The type of metadata to be read/written
+ */
+public class MetadataManager<T> {
```

Comment:
`MetadataManager` more or less sounds like a `MetadataParser` and the existing `MetadataParser` should be probably `MetadataContentParser`.

Reply:
Makes sense, thanks for pointing this out. This will also help avoid "Manager" in the name which increases the scope of this construct and can get bloated with functionality in future.